spec/features/favorites/add_favorite_pet_spec.rb

```diff
@@ -0,0 +1,28 @@
+require 'rails_helper'
+
+RSpec.describe "Add Favorite Pet" do
+  describe "as a visitor when visiting a pet's show page" do
+    it "I can add it to my favorites list" do
+      shelter = Shelter.create(name: "Happy Puppies", address: "55 Street St", city: "Danger Mountain", state: "UT", zip: "80304")
+      pet_1 = Pet.create(image: "image.jpeg", name: "Kunga", approximate_age: "1", sex: "male", shelter_id: shelter.id)
+      pet_2 = Pet.create(image: "image.jpeg", name: "Honey Pie", approximate_age: "11", sex: "female", shelter_id: shelter.id)
+
+      visit "/pets/#{pet_1.id}"
+
+      expect(page).to have_button("Add to Favorites")
+
+      click_button("Add to Favorites")
```

Comment:
I think that click_button verifies that we have a button so the expect(page).to have_button isn't necessary, but also it doesn't hurt anything to put it there!